awssql/plugins/bg/bg_status_provider.go

```diff
@@ -0,0 +1,982 @@
+			nil,
+		))
+
+		interimStatus := b.interimStatuses[role.GetValue()]
+		if interimStatus.IsZero() {
+			continue
+		}
+
+		connectRouting = append(connectRouting, NewSubstituteConnectRouting(
+			host_info_util.GetHostAndPort(host, interimStatus.Port),
+			role,
+			blueIpHostInfo,
+			[]*host_info_util.HostInfo{blueHostInfo},
+			nil,
+		))
+	}
+	return connectRouting
+}
+
+func (b *BlueGreenStatusProvider) GetStatusOfInProgress() driver_infrastructure.BlueGreenStatus {
+	if b.IsSwitchoverTimerExpired() {
+		slog.Debug(error_util.GetMessage("BlueGreenDeployment.switchoverTimeout"))
+		if b.rollback {
+			return b.GetStatusOfCreated()
+		}
+		return b.GetStatusOfCompleted()
+	}
+
+	var connectRouting []driver_infrastructure.ConnectRouting
+	if b.suspendNewBlueConnectionsWhenInProgress {
+		// All blue and green connect calls should be suspended.
+		connectRouting = []driver_infrastructure.ConnectRouting{NewSuspendConnectRouting("", driver_infrastructure.SOURCE, b.bgdId)}
+	} else {
+		// If we're not suspending new connections then, at least, we need to use IP addresses.
+		connectRouting = b.AddSubstituteBlueWithIpAddressConnectRouting()
+	}
+
+	connectRouting = append(connectRouting, NewSuspendConnectRouting("", driver_infrastructure.TARGET, b.bgdId))
+
+	// All connect calls with IP address that belongs to blue or green host should be suspended.
+	uniqueIpAddresses := make(map[string]bool, b.hostIpAddresses.Size())
+	for _, ipAddress := range b.hostIpAddresses.GetAllEntries() {
+		if _, ok := uniqueIpAddresses[ipAddress]; ipAddress != "" && !ok {
+			uniqueIpAddresses[ipAddress] = true
+			if b.suspendNewBlueConnectionsWhenInProgress {
+				interimStatus := b.interimStatuses[driver_infrastructure.SOURCE.GetValue()]
+				if !interimStatus.IsZero() && utils.FilterMapFindFirstValue(interimStatus.StartIpAddressesByHostMap, func(s string) bool {
+					return s != "" && s == ipAddress
+				}) != "" {
+					connectRouting = append(connectRouting, NewSuspendConnectRouting(ipAddress, driver_infrastructure.BlueGreenRole{}, b.bgdId))
+					connectRouting = append(connectRouting, NewSuspendConnectRouting(host_info_util.GetHostAndPort(ipAddress, interimStatus.Port),
+						driver_infrastructure.BlueGreenRole{}, b.bgdId))
+					continue
+				}
+			}
+			// Try to confirm that the ipAddress belongs to one of the green hosts
+			interimStatus := b.interimStatuses[driver_infrastructure.TARGET.GetValue()]
+			if !interimStatus.IsZero() && utils.FilterMapFindFirstValue(interimStatus.StartIpAddressesByHostMap, func(s string) bool {
+				return s != "" && s == ipAddress
+			}) != "" {
+				connectRouting = append(connectRouting, NewSuspendConnectRouting(ipAddress, driver_infrastructure.BlueGreenRole{}, b.bgdId))
+				connectRouting = append(connectRouting, NewSuspendConnectRouting(host_info_util.GetHostAndPort(ipAddress, interimStatus.Port),
+					driver_infrastructure.BlueGreenRole{}, b.bgdId))
+				continue
+			}
+		}
+	}
```

Comment:
This is a bit hard to follow with the nested conditions, is it possible to simplify this even if it goes through the list of ip addresses? 
1. Create a set of unique Ip addresses first (through a helper method), seems like we only do the status with unique ip addresses
2. Create a helper method that will append to connectRouting
3. Create additional helper methods (if needed) to simplify the logic of the conditions like `ipAddressBelongsToGreenHost`